sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/CosmosSourceConnector.java

```diff
@@ -215,12 +215,28 @@ private boolean shouldCreateMetadataContainerIfNotExists() {
     }
 
     private Mono<CosmosContainerResponse> createMetadataContainer() {
+        return this.createMetadataContainer(METADATA_CONTAINER_DEFAULT_RU_CONFIG)
+            .onErrorResume(throwable -> {
+                // for serverless database account, creating container with throughput configured will get 400/0 exceptions
+                if (KafkaCosmosExceptionsHelper.isBadRequestException(throwable)) {
```

Comment:
Can bad request exception happen because of any other reason? If so, is it still worth creating the container?

Reply:
it could be - but for the serverless account error - currently there is no explicit error code matching to it, so I think we still should try it one more time

Updated to also check the substrates code being 0 to exclude some other 400 errors